src/main/java/io/cdap/plugin/gcp/bigquery/source/CdapBigQueryInputFormat.java

```diff
@@ -0,0 +1,173 @@
+package io.cdap.plugin.gcp.bigquery.source;
+
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.cloud.hadoop.io.bigquery.AbstractBigQueryInputFormat;
+import com.google.cloud.hadoop.io.bigquery.AvroRecordReader;
+import com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration;
+import com.google.cloud.hadoop.io.bigquery.BigQueryHelper;
+import com.google.cloud.hadoop.io.bigquery.BigQueryStrings;
+import com.google.cloud.hadoop.io.bigquery.Export;
+import com.google.cloud.hadoop.io.bigquery.ExportFileFormat;
+import com.google.cloud.hadoop.io.bigquery.NoopFederatedExportToCloudStorage;
+import com.google.cloud.hadoop.io.bigquery.ShardedExportToCloudStorage;
+import com.google.cloud.hadoop.io.bigquery.UnshardedExportToCloudStorage;
+import com.google.cloud.hadoop.util.ConfigurationUtil;
+import com.google.cloud.hadoop.util.HadoopToStringUtil;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import org.apache.avro.generic.GenericData;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * BigQuery input format, splits query from the configuration into list of queries
+ * in order to create input splits.
+ */
+public class CdapBigQueryInputFormat extends AbstractBigQueryInputFormat<LongWritable, GenericData.Record> {
```

Comment:
It looks like there's a lot of copied code here and it's hard for me to tell what is copied and what is ours.

It's simpler to decorate the bigquery inputformat rather than trying to hook into the Export stuff. From our perspective, all we're doing is running a query that reads relevant partitions into a temporary table, then configuring the bigquery inputformat to read from that temporary table. So it should look something like:

```
public class CdapBigQueryInputFormat extends InputFormat<LongWritable, GenericData.Record> {
  
  public List<InputSplit> getSplits(JobContext context) {
    // code to execute query that writes to temporary table
    ...
    // set tablename configuration setting to temporary table
    return delegate.getSplits(context);
  }
}
```

Reply:
InputFormat class was refactored.